pandas-stubs/core/frame.pyi

```diff
@@ -608,10 +611,35 @@ class DataFrame(NDFrame, OpsMixin):
         self, expr: _str, *, inplace: Literal[False] = ..., **kwargs
     ) -> DataFrame: ...
     def eval(self, expr: _str, *, inplace: _bool = ..., **kwargs): ...
+    AstypeArgExt: TypeAlias = (
+        AstypeArg
+        | Literal[
+            "number",
+            "datetime64",
+            "datetime",
+            "timedelta",
+            "timedelta64",
+            "datetimetz",
+            "datetime64[ns]",
+        ]
+    )
+    AstypeArgExtList: TypeAlias = AstypeArgExt | list[AstypeArgExt]
```

Comment:
I think if you added the following here as the first 2 overloads, you could catch the case of the string arguments not being allowed:
```python
@overload
def select_dtypes(self, include: StrDtypeArg, exclude: AstypeArgExtList | None = ...) -> Never: ...
@overload
def select_dtypes(self, include: AstypeArgExtList | None = ..., exclude: StrDtypeArg) -> Never: ...
```
And if you added this, I think that you'd catch the case of 2 empty lists:
```python
@overload
def select_dtypes(self, include: list[Never], exclude: list[Never]) -> Never: ...
```
You import `Never` from `typing_extensions`

There's a possibility that either mypy and/or pyright will complain about overlapping overloads, but you can add `#type: ignore` and `# pyright: ignore` as needed in the stubs to ignore that complaint.

Reply:
That is implemented. `Never` is actually stronger than I expected since pylance read the stubs and visually show that the program will stop there.